internal/pkg/plugininstaller/reposcaffolding/utils.go

```diff
@@ -0,0 +1,68 @@
+package reposcaffolding
+
+import (
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/devstream-io/devstream/pkg/util/github"
+	"github.com/devstream-io/devstream/pkg/util/log"
+)
+
+func downloadGithubRepo(org, repo, workpath string) error {
+	ghOption := &github.Option{
+		Org:      org,
+		Repo:     repo,
+		NeedAuth: false,
+		WorkPath: workpath,
+	}
+	ghClient, err := github.NewClient(ghOption)
+	if err != nil {
+		return err
+	}
+
+	if err = ghClient.DownloadLatestCodeAsZipFile(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func replaceAppNameInPathStr(filePath, appName string) (string, error) {
+	if !strings.Contains(filePath, appNamePlaceHolder) {
+		return filePath, nil
+	}
+
+	reg, err := regexp.Compile(appNamePlaceHolder)
+	if err != nil {
+		return "", err
+	}
+	newFilePath := reg.ReplaceAllString(filePath, appName)
```

Comment:
If the content of the regex is provided by the program, it can be written like this if it is guaranteed to be correct:

```go
newFilePath := regexp.MustCompile(appNamePlaceHolder).ReplaceAllString(filePath, appName)
```